common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1874,6 +1874,14 @@ def _update_item_from_fields(
                 block_data.fields = settings
 
                 new_id = new_structure['_id']
+
+                # When a draft block from a published structure is reverted in revert_to_published(),
+                # the update_version is updated while source_version is not. If source_version is not None,
+                # set it to None - so that has_changes() will use update_version for comparison and return True.
+                # Prevents the block from getting stuck in the published state.
+                if block_data.edit_info.source_version:
+                    block_data.edit_info.source_version = None
```

Comment:
This comment reads as though the change is being made to compensate for a flaw in another piece of code. Why is `update_version` changed but `source_version` isn't changed? Is that the correct behavior?

Also, since this is only setting `source_version` if `source_version` is truthy, maybe it should just always set `source_version` to `None`? That might be clearer.